src/vs/workbench/services/remote/common/remoteExplorerService.ts

```diff
@@ -74,6 +74,7 @@ export function mapHasTunnelLocalhostOrAllInterfaces(map: Map<string, Tunnel>, h
 		if (otherHost) {
 			return mapHasTunnel(map, otherHost, port);
 		}
+		return false;
```

Comment:
At this point, we have checked and the map does not in fact have the tunnel already and we should return false.